android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java

```diff
@@ -462,29 +462,102 @@ public static boolean executeHardwareKeystroke(
   }
 
   /**
-   * Handle the globe key longpress action. Currently just for INAPP only
+   * Handle the globe key action on the lock screen
+   * @param context
+   */
+  private static void doGlobeKeyLockscreenAction(Context context) {
+    // Override globe key action if screen is locked:
+    // 1. Switch to next Keyman keyboard (no menu)
+    // 2. When all the Keyman keyboards have been cycled through, advance to the next system keyboard
+    if (sysKbStartingIndexOnLockScreen == getCurrentKeyboardIndex(context)) {
```

Comment:
Does this variable need to be called `sysKbStartingIndexOnLockScreen` now? Or can we rename to `startingIndexOnLockScreen` (even that could be better, e.g. `startingKeyboardIndexOnLockScreen`)?